src/v2/components/ChannelContents/components/ChannelContentsItem/index.tsx

```diff
@@ -24,64 +64,114 @@ interface Props {
   }: {
     oldIndex: number
     newIndex: number
-  }) => any
+  }) => void
+  onItemIntersected: (index: number) => void
 }
 
 export const ChannelContentsItem: React.FC<Props> = memo(
-  ({ channel, connectable, index, context, onRemove, onChangePosition }) => {
+  ({
+    channel,
+    connectable,
+    index,
+    context,
+    onRemove,
+    onChangePosition,
+    onItemIntersected,
+  }) => {
     const [isHovering, setHover] = useState(false)
 
-    const startHover = useCallback(() => {
-      if (isTouchDevice()) return false
+    type Ev = (e: React.MouseEvent<HTMLDivElement, MouseEvent>) => void
+    const startHover = useCallback<Ev>(() => {
+      if (isTouchDevice()) return
       setHover(true)
     }, [])
-    const endHover = useCallback(() => {
-      if (isTouchDevice()) return false
+    const endHover = useCallback<Ev>(() => {
+      if (isTouchDevice()) return
       setHover(false)
     }, [])
 
     const handleChangePosition = useCallback(
-      (newIndex: number) => onChangePosition({ oldIndex: index, newIndex }),
+      (newIndex: number) => {
+        onChangePosition({
+          oldIndex: index,
+          newIndex,
+        })
+      },
       [index, onChangePosition]
     )
 
+    const intersectionObserverCallback = useCallback<
```

Comment:
There's a pretty cool optimization we can make later, where we only call `onItemIntersected` if the element is tracked as being in view for a duration of time rather than immediately.

The end result will be that a block's page will only get queried if a block is in view for ~0.2 seconds instead of immediately. This will reduce the amount of queries in the event that a user is quickly scrolling down to the bottom of a grid, because we won't be loading the intermediate pages since they would only briefly be in view.